### PR TITLE
Add SingleStoreReader

### PR DESCRIPTION
**Pull request by apeng-singlestore**
Loads documents from a singlestore table that has been populated with text and embedding columns

Fixed the poetry.lock issue

### CODE REVIEW

File: llama_hub/singlestore/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# SingleStore Loader
+
+The SingleStore Loader retrieves a set of documents from a specified table in a SingleStore database. The user initializes the loader with database information and then provides a search embedding for retrieving similar documents.
+
+## Usage
+
+Here's an example usage of the SingleStoreReader:
+
+```python
+from llama_index import download_loader
```
Comment (jerryjliu) [inline llama_hub/singlestore/README.md:10]: nit: we're now officially encouraging people to use llama-hub as a package: `from llama_hub.singlestore.base import SingleStoreReader`

File: llama_hub/singlestore/base.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self.user = user
+        self.password = password
+        self.dbname = dbname
+        self.table_name = table_name
+        self.content_field = content_field
+        self.vector_field = vector_field
+
+        pymysql.install_as_MySQLdb()
+
+        self.DatabaseReader = download_loader('DatabaseReader')
```
Comment (jerryjliu) [inline llama_hub/singlestore/base.py:41]:
```
try:
    from llama_hub.utils import import_loader
    
    PandasCSVReader = import_loader("PandasCSVReader")
except ImportError:
    PandasCSVReader = download_loader("PandasCSVReader")
```

Comment (apeng-singlestore) [reply llama_hub/singlestore/base.py:41]: Is there a reason for using `PandasCSVReader` instead of `DatabaseReader`?

Comment (jerryjliu) [reply llama_hub/singlestore/base.py:41]: oh my bad, that was just an example

File: llama_hub/singlestore/base.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+"""SingleStore reader."""
+
+from typing import Any, Dict, List, Optional
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+from llama_index import download_loader, ListIndex
+
+import pymysql
```
Comment (Disiok) [inline llama_hub/singlestore/base.py:9]: Please do the imports from within the class and raise import error with appropriate error messages. 

For example: https://github.com/emptycrown/llama-hub/blob/main/llama_hub/couchdb/base.py

**Review by Disiok — CHANGES_REQUESTED** [review]
Should be a very minor change to pass CI tests!

**Review by Disiok — APPROVED** [review]
LGTM!

**Pull request merged**